Source/Core/InputCommon/ControllerInterface/Xlib/XInput2.cpp

```diff
@@ -2,7 +2,10 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#pragma clang diagnostic push
```

Comment:
Could you replace `clang` with `GCC`? GCC 7 also warns about `register`:

>-Wregister warns about uses of the register storage specifier. In C++17 this keyword has been removed and for C++17 this is a pedantic warning enabled by default. The warning is not emitted for the GNU Explicit Register Variables extension.

`#pragma clang` only works with clang, whereas `#pragma GCC` works with both clang and GCC.

Reply:
yeah sure